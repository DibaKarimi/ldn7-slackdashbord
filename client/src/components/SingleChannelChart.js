import { Bar, defaults } from "react-chartjs-2";

defaults.font.family = "Varela Round";

const SingleChannelChart = (props) => {
	const state = {
		type: "bar",
		data: {
			labels: ["Previous Week", "Current Week"],
			datasets: [
				{
					label: "Messages",
					backgroundColor: "#ffa600",
					borderColor: "#003f5c",
					borderWidth: 1,
					hoverBackgroundColor: "#ff7c43",
					hoverBorderColor: "#003f5c",
					data: props.messagesDataSet,
				},
				{
					label: "Reactions",
					backgroundColor: "#ff6361",
					borderColor: "#003f5c",
					borderWidth: 1,
					hoverBackgroundColor: "#d45087",
					hoverBorderColor: "#003f5c",
					data: props.reactionsDataSet,
				},
			],
		},
		options: {
			maintainAspectRatio: true,
			aspectRatio: 2,
			xAxes: [
				{
					scaleLabel: {
						display: true,
						labelString: "Weeks",
					},
				},
			],
			yAxes: [
				{
					ticks: {
						beginAtZero: true,

						labelString: "Totals",
					},
				},
			],
		},
	};

	return (
		<div className="singleChannelChart">
			<Bar data={state.data} />
		</div>
	);
};

export default SingleChannelChart;
