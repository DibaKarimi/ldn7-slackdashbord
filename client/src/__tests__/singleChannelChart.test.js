import React from "react";
import SingleChannelChart from "../components/SingleChannelChart";

import { shallow } from "enzyme";

describe("SingleChannelChart", () => {
	it("should render chart for a single channel correctly", () => {
		const component = shallow(<SingleChannelChart />);

		expect(component).toMatchSnapshot();
	});
});
