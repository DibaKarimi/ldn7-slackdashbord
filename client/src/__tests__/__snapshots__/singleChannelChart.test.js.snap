// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleChannelChart should render chart for a single channel correctly 1`] = `
<div
  className="chart"
>
  <ForwardRef
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "#ffa600",
            "borderColor": "#003f5c",
            "borderWidth": 1,
            "data": undefined,
            "hoverBackgroundColor": "#ff7c43",
            "hoverBorderColor": "#003f5c",
            "label": "Messages",
          },
          Object {
            "backgroundColor": "#ff6361",
            "borderColor": "#003f5c",
            "borderWidth": 1,
            "data": undefined,
            "hoverBackgroundColor": "#d45087",
            "hoverBorderColor": "#003f5c",
            "label": "Reactions",
          },
        ],
        "labels": Array [
          "Current Week",
          "Previous Week",
        ],
      }
    }
    options={
      Object {
        "legend": Object {
          "display": false,
        },
        "responsive": true,
        "type": "bar",
      }
    }
  />
</div>
`;
